base/abstractarraymath.jl

```diff
@@ -518,6 +518,9 @@ function check(arr, inner, outer)
         # TODO: Currently one based indexing is demanded for inner !== nothing,
         # but not for outer !== nothing. Decide for something consistent.
         Base.require_one_based_indexing(arr)
+        if !all(n -> n isa Integer, inner)
+            throw(ArgumentError("repeat requires integer counts, got inner = $inner"))
```

Comment:
Perhaps this error message may be a lazy string

Reply:
Could be. For now I left it just like all others in this check function.

Reply:
I imagine this branch will DCE much of the time